subprojects/core/src/main/java/org/gradle/api/internal/DefaultMutationGuard.java

```diff
@@ -19,12 +19,13 @@
 import org.gradle.api.Action;
 
 public class DefaultMutationGuard extends AbstractMutationGuard {
-    private final ThreadLocal<Boolean> mutationGuardState = new ThreadLocal<Boolean>() {
+    private final ThreadLocal<Boolean> mutationGuardState = new GuardState();
```

Comment:
`ThreadLocal.withInitial(...);` is no good here because CC can't serialize the Supplier.